processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -121,7 +124,10 @@ public SegmentMetadataQueryQueryToolChest(SegmentMetadataQueryConfig config, Gen
       @Override
       protected Ordering<SegmentAnalysis> makeOrdering(Query<SegmentAnalysis> query)
```

Comment:
There seem to be no point for this anonymous QueryRunner to extend `ResultMergeQueryRunner` rather than `BySegmentSkippingQueryRunner` directly, because it overrides `doRun()`. Then this QueryRunner doesn't need to follow API of `ResultMergeQueryRunner`, it may accept `SegmentMetadataQuery` instead of `Query<SegmentAnalysis>` in `makeOrdering()` and `createMergeFn()`, to avoid casting.